### PR TITLE
infoschema: improve the security vunerability of statistics infoschema table

### PR DESCRIPTION
**Pull request by reafans**
### What problem does this PR solve? 
improve the security vunerability of statistics infoschema table

### What is changed and how it works?
before：root and new user both can access the statistics table
```
mysql -u root -h 127.0.0.1 -P 4000
mysql> use information_schema;
mysql> select * from statistics limit 3;
+---------------+--------------+--------------+------------+--------------+------------+--------------+-------------+-----------+-------------+----------+--------+----------+------------+---------+---------------+
| TABLE_CATALOG | TABLE_SCHEMA | TABLE_NAME   | NON_UNIQUE | INDEX_SCHEMA | INDEX_NAME | SEQ_IN_INDEX | COLUMN_NAME | COLLATION | CARDINALITY | SUB_PART | PACKED | NULLABLE | INDEX_TYPE | COMMENT | INDEX_COMMENT |
+---------------+--------------+--------------+------------+--------------+------------+--------------+-------------+-----------+-------------+----------+--------+----------+------------+---------+---------------+
| def           | mysql        | columns_priv | 0          | mysql        | PRIMARY    |            1 | Host        | A         |        NULL |     NULL | NULL   |          | BTREE      |         |               |
| def           | mysql        | columns_priv | 0          | mysql        | PRIMARY    |            2 | DB          | A         |        NULL |     NULL | NULL   |          | BTREE      |         |               |
| def           | mysql        | columns_priv | 0          | mysql        | PRIMARY    |            3 | User        | A         |        NULL |     NULL | NULL   |          | BTREE      |         |               |
+---------------+--------------+--------------+------------+--------------+------------+--------------+-------------+-----------+-------------+----------+--------+----------+------------+---------+---------------+
3 rows in set (0.00 sec)

mysql> create user reafans;
mysql -u reafans -h 127.0.0.1 -P 4000
mysql> use information_schema;
mysql> select * from statistics limit 2;
+---------------+--------------+--------------+------------+--------------+------------+--------------+-------------+-----------+-------------+----------+--------+----------+------------+---------+---------------+
| TABLE_CATALOG | TABLE_SCHEMA | TABLE_NAME   | NON_UNIQUE | INDEX_SCHEMA | INDEX_NAME | SEQ_IN_INDEX | COLUMN_NAME | COLLATION | CARDINALITY | SUB_PART | PACKED | NULLABLE | INDEX_TYPE | COMMENT | INDEX_COMMENT |
+---------------+--------------+--------------+------------+--------------+------------+--------------+-------------+-----------+-------------+----------+--------+----------+------------+---------+---------------+
| def           | mysql        | columns_priv | 0          | mysql        | PRIMARY    |            1 | Host        | A         |        NULL |     NULL | NULL   |          | BTREE      |         |               |
| def           | mysql        | columns_priv | 0          | mysql        | PRIMARY    |            2 | DB          | A         |        NULL |     NULL | NULL   |          | BTREE      |         |               |
+---------------+--------------+--------------+------------+--------------+------------+--------------+-------------+-----------+-------------+----------+--------+----------+------------+---------+---------------+
2 rows in set (0.00 sec)
```
after：root can access the table but new user can only get the empty set.
```
mysql -u root -h 127.0.0.1 -P 4000
mysql> use information_schema;
mysql> select * from statistics limit 2;
+---------------+--------------+--------------+------------+--------------+------------+--------------+-------------+-----------+-------------+----------+--------+----------+------------+---------+---------------+
| TABLE_CATALOG | TABLE_SCHEMA | TABLE_NAME   | NON_UNIQUE | INDEX_SCHEMA | INDEX_NAME | SEQ_IN_INDEX | COLUMN_NAME | COLLATION | CARDINALITY | SUB_PART | PACKED | NULLABLE | INDEX_TYPE | COMMENT | INDEX_COMMENT |
+---------------+--------------+--------------+------------+--------------+------------+--------------+-------------+-----------+-------------+----------+--------+----------+------------+---------+---------------+
| def           | mysql        | columns_priv | 0          | mysql        | PRIMARY    |            1 | Host        | A         |        NULL |     NULL | NULL   |          | BTREE      |         |               |
| def           | mysql        | columns_priv | 0          | mysql        | PRIMARY    |            2 | DB          | A         |        NULL |     NULL | NULL   |          | BTREE      |         |               |
+---------------+--------------+--------------+------------+--------------+------------+--------------+-------------+-----------+-------------+----------+--------+----------+------------+---------+---------------+
2 rows in set (0.00 sec)

mysql -u reafans -h 127.0.0.1 -P 4000
mysql> use information_schema;
mysql> select * from statistics limit 2;
Empty set (0.00 sec)
```

### Check List 

Tests 

 - Integration test

Code changes

 - Has exported function/method change

Side effects
None

Related changes
None

### CODE REVIEW

File: infoschema/tables.go [hunk 1]
```diff
@@ -2375,7 +2380,7 @@ func (it *infoschemaTable) getRows(ctx sessionctx.Context, cols []*table.Column)
 	case tableColumns:
 		fullRows = dataForColumns(ctx, dbs)
 	case tableStatistics:
-		fullRows = dataForStatistics(dbs)
+		fullRows = dataForStatistics(ctx, dbs)
```
Comment (crazycs520) [inline infoschema/tables.go]: Check other tables too?
- `analyze_status`
- `key_column_usage`
- `schemata`
- `statistics`
- `table_constraints`
- `tikv_region_peers`
- `tikv_region_status`

Comment (reafans) [reply infoschema/tables.go]: OK

**Comment by zz-jason** [comment]
@reafans please add some proper labels for this PR.

**Comment by reafans** [comment]
> @reafans please add some proper labels for this PR.

It seems that there is no label for component/infoschema

**Comment by tiancaiamao** [comment]
Any test to cover the change? @reafans

**Comment by reafans** [comment]
> Any test to cover the change? @reafans

added test，PTAL @tiancaiamao

File: infoschema/tables_test.go [hunk 2]
```diff
@@ -449,7 +449,17 @@ func (s *testTableSuite) TestSomeTables(c *C) {
...
+	//test the privilege of new user for information_schema
+	tk.MustExec("create user statistics_tester")
+	tk.MustExec("flush privileges")
+	tk1 := testkit.NewTestKit(c, s.store)
+	tk1.MustExec("use information_schema")
+	c.Assert(tk1.Se.Auth(&auth.UserIdentity{
+		Username: "statistics_tester",
+		Hostname: "127.0.0.1",
+	}, nil, nil), IsTrue)
+	tk1.MustQuery("select * from information_schema.STATISTICS;").Check([][]interface{}{})
```
Comment (Deardrops) [inline infoschema/tables_test.go]: Suggestion:
Adding one more test case, grant necessary roles for new user, and it should get non-empty result.

Comment (tiancaiamao) [reply infoschema/tables_test.go]: PTAL @imtbkcat

Comment (imtbkcat) [reply infoschema/tables_test.go]: @reafans try this case:
```
create role r_schema;
grant all privileges on *.* to r_schema;
grant r_schema to statistics_tester;
```
then create a session as `statistics_tester` and run query below.

```
set role r_schema;
select * from information_schema.STATISTICS;
```
it should get non-empty result.

Comment (reafans) [reply infoschema/tables_test.go]: done，PTAL

**Pull request merged**